cmd/metacache-stream.go

```diff
@@ -817,36 +816,62 @@ func (w *metacacheBlockWriter) Close() error {
 type metacacheBlock struct {
 	data  []byte
 	n     int
-	First string `json:"f"`
-	Last  string `json:"l"`
-	EOS   bool   `json:"eos,omitempty"`
+	first string
+	last  string
+	eos   bool
+}
+
+const metacacheBlockReservedHeaderPrefix = ReservedMetadataPrefixLower + "metacache-part-"
+
+// This code is specifically written in this form to avoid json unmarshalling
+//
+// BenchmarkKVSimple-4     13431099                88.15 ns/op           48 B/op          1 allocs/op
+// v/s
+// BenchmarkKVJSON-4        1510602               789.4 ns/op           224 B/op          4 allocs/op
+//
+// So that only allocation this function ever does is for strings.SplitN()
+func (b *metacacheBlock) loadHeaderKV(meta map[string]string, partKey string) error {
```

Comment:
Breaking change. `metacacheStreamVersion` must be bumped and Storage API versions. This will dump all caches.

Reply:
It will because it's fine since we purge all caches upon upgrade. So no version migration needed @klauspost